lib/commands/performance.js

```diff
@@ -0,0 +1,339 @@
+import { errors } from 'appium-base-driver';
+import { sleep } from 'asyncbox';
+import _ from 'lodash';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.postDismissAlert = async function () {
+  throw new errors.NotYetImplementedError();
+};
+
+const SUPPORTED_PERFORMANCE_DATA_TYPES = {
+  cpuinfo: 'the amount of cpu by user and kernel process - cpu information for applications on real devices and simulators',
+  memoryinfo: 'the amount of memory used by the process - memory information for applications on real devices and simulators',
+  batteryinfo: 'the remaining battery power - battery power information for applications on real devices and simulators',
+  networkinfo: 'the network statistics - network rx/tx information for applications on real devices and simulators'
+};
+
+//
+// returns the information type of the system state which is supported to read as like cpu, memory, network traffic, and battery.
+// output - array like below
+//[cpuinfo, batteryinfo, networkinfo, memoryinfo] 
+//
+commands.getPerformanceDataTypes = function () {
+  return _.keys(SUPPORTED_PERFORMANCE_DATA_TYPES);
+};
+
+// returns the information type of the system state which is supported to read as like cpu, memory, network traffic, and battery.
+//input - (packageName) the package name of the application 
+//        (dataType) the type of system state which wants to read. It should be one of the keys of the SUPPORTED_PERFORMANCE_DATA_TYPES
+//        (dataReadTimeout) the number of attempts to read
+// output - table of the performance data, The first line of the table represents the type of data. The remaining lines represent the values of the data.
+//
+// in case of battery info : [[power], [23]]
+// in case of memory info : [[totalPrivateDirty, nativePrivateDirty, dalvikPrivateDirty, eglPrivateDirty, glPrivateDirty, totalPss, nativePss, dalvikPss, eglPss, glPss, nativeHeapAllocatedSize, nativeHeapSize], [18360, 8296, 6132, null, null, 42588, 8406, 7024, null, null, 26519, 10344]]
```

Comment:
ok, I understand what you mean. 

the reason why I defined the return type like that 

1. 
for memory, cpu, and battery, there could have a value for a key.
but for network, there could have many values for a key.
So, There is the key duplication.

2. 
User can use the result data as the type of Array in Java.
so, user does not need to add the library for the type of json.

3. 
for the network statistics,
The key is different between real device and emulator.
I have not tested many devices and emulator, I think there might be some device or emulator which has the different key. 
so, I made the key as variable not constant.
I think It is easy to manage the code for like this variations.

I thank you for your recommendation.
In Korea, It is difficult to get the development culture like this.
Anyway, If you still think that you prefer the json type than the Array, then I will change the code.